src/gcinfo/gcinfodumper.cpp

```diff
@@ -300,7 +300,7 @@ PORTABILITY_ASSERT("GcInfoDumper::ReportPointerRecord is not implemented on this
                 pReg = *(SIZE_T**)(pContext + rgRegisters[iReg].cbContextOffset);
                 if (iEncodedReg == iSPRegister)
                 {
-                    pReg = (SIZE_T*)((BYTE*)pRD->pCurrentContext + rgRegisters[iEncodedReg].cbContextOffset);
+                    pReg = (SIZE_T*)((BYTE*)pRD->pCurrentContext + rgRegisters[iReg].cbContextOffset);
```

Comment:
ok, what i actually meant was the iSPRegister from this function's point of view should be 30. but as it is the function is hacky and we use iReg == 30 to find that we are at iSPRegister, Your solution is also equally good.